File: 이것이 취업을 위한 코딩 테스트다/python/모험가 길드.py
n = int(input())
data = list(map(int, input().split()))
data.sort()  # 오름차순 정렬

#  1 2 2 2 4

result = 0
ctr = 0

for i in data:
  ctr += 1
  if ctr >= i:
    result += 1
    ctr = 0

print(result)

File: 이것이 취업을 위한 코딩 테스트다/c++/모험가 길드.cpp
#include <algorithm>
#include <iostream>
using namespace std;

int main() {
  int n, arr[9999];
  int result = 0, ctr = 0;
  cin >> n;

  // cin은 split 따위를 쓸필요가 없다.
  for (int i = 0; i < n; i++) {
    cin >> arr[i];
  }

  sort(arr, arr + n); // ASC

  for (int i = 0; i < n; i++) {
    ctr++;
    if (ctr >= arr[i]) {
      result++;
      ctr = 0;
    }
  }

  cout << result << endl;
}